src/pgcopy/src/copy.rs

```diff
@@ -0,0 +1,769 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
```

Comment:
virtually everything in this file is moved from `pgwire::message`. 

The only change is swapping the `ErrorResponse` for the local `CopyErrorNotSupportedResponse`, in order to avoid a dependency on `rust-postgres` just for the error type.

Reply:
^ Not quite true any more, I added a `RawIterator` that has a more ergonomic way of parsing out each value